travis: Integrate test coverage

kaushik-rohit opened the pull request:
This commit integrate coverage to travis. Test coverage
provides us a statistical information about how much
codebase is being currently tested or covered by the test
we have.

Issue link: https://github.com/prasadtalasila/IRCLogParser/issues/191

## What? Why?
Fix #91 #95 

Changes proposed in this pull request:
- Integrate test coverage into travis

## Checks
- [x] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [x] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Any images?

## Notify reviewers
@rohangoel96

kaushik-rohit commented:
Below is the result of code coverage run.

Name Stmts Miss Cover Missing
lib/analysis/channel.py 226 217 4% 25-160, 182-325, 341-351, 376-400
lib/analysis/community.py 61 61 0% 1-108
lib/analysis/network.py 450 106 76% 196-197, 201-215, 253-254, 259-264, 269-288, 374-382, 398-427, 514-520, 594, 740-789
lib/analysis/user.py 270 99 63% 81, 122-127, 131, 217-218, 264-267, 295-420
lib/in_out/reader.py 37 4 89% 45-46, 53-54
lib/in_out/saver.py 55 55 0% 1-129
lib/nickTracker.py 109 3 97% 98, 107, 157
lib/scummvm/nickTracker.py 109 109 0% 1-158
lib/slack/analysis/channel.py 226 226 0% 1-400
lib/slack/analysis/network.py 448 448 0% 1-787
lib/slack/in_out/reader.py 34 34 0% 1-69
lib/slack/nickTracker.py 109 109 0% 1-158
lib/slack/util.py 95 95 0% 1-255
lib/util.py 95 19 80% 67, 128-131, 221, 228-233, 242-244, 252-255
lib/validate.py 6 6 0% 1-33
lib/vis.py 213 213 0% 1-487
sample.py 11 11 0% 1-11
scummvm.py 144 144 0% 1-226
setup.py 19 19 0% 1-25
slack-gc.py 146 146 0% 1-230
slack.py 71 71 0% 1-108
test/tests.py 153 14 91% 40, 115-124, 132-137
ubuntu.py 225 225 0% 1-303

TOTAL 3312 2434 27%

prasadtalasila commented on .travis.yml:
```diff
@@ -16,6 +16,7 @@ python:
 install:
   - pip install -U pip wheel
   - pip install sphinx
+  - pip install coverage
```
Please add the relevant commands to [install.sh](https://github.com/prasadtalasila/IRCLogParser/blob/dev/install.sh) as well so that we can run the coverage test on development machines as well.

prasadtalasila commented on .travis.yml:
```diff
@@ -32,6 +33,8 @@ script:
 
 after_success:
   # - coverage run test/tests.py && codeclimate-test-reporter --file .coverage --token $CODECLIMATE_REPO_TOKEN
+  - coverage run --source=. -m unittest discover -s test/
```
It might be a good idea to put these commands in a script/test bash file and invoke that file in travis.yml. Done that way, we can also use the script during development as well.

prasadtalasila replied:
Please see the conventions of GitHub's [one script to rule them all](https://github.com/github/scripts-to-rule-them-all) for reference.

kaushik-rohit replied:
install.sh should also move to script then.

prasadtalasila replied:
Yup. That would be appropriate. Also we need to rename it to install (just to stick to naming conventions of GitHub)

prasadtalasila reviewed (changes requested):
@kaushik-rohit please make the suggested changes.

prasadtalasila commented on script/test.sh:
```diff
@@ -0,0 +1,2 @@
+coverage run --source=. -m unittest discover -s ../test/
```
@kaushik-rohit The file name would just be **test**

prasadtalasila commented on install.sh:
```diff
@@ -16,6 +16,7 @@ apt-get install -y python-scipy
 apt-get install libcairo2-dev
 apt-get install libffi-dev
 pip install cairocffi
+pip install coverage
```
Please double check the command on a server machine.

kaushik-rohit replied:
Sure, will do that on Monday. I am not on campus.

prasadtalasila reviewed (changes requested):
@kaushik-rohit please make the suggested changes.

prasadtalasila commented:
@kaushik-rohit Please put in travis integration for code climate test coverage reporter as well. That task would fit in nicely with this PR. We already have the code climate token, hence everything is set from the authorization point of view. You just need to put in the right command in travis.yml.

kaushik-rohit commented:
@prasadtalasila Sir, where is the $CODECLIMATE_REPO_TOKEN located?

prasadtalasila commented:
The CODECLIMATE_REPO_TOKEN is set as an environment variable in the travis build setup. You don't need to set it up manually.

kaushik-rohit committed:
tests: Improve unittest and integrate code coverage

1) renaming test directory and modules and add __init__.py, so that
tests are discoverable by unittest.

2) integrate coverage in travis

3) add .coverage to gitignore

prasadtalasila commented on tests/.coverage:
```diff
@@ -0,0 +1 @@
+!coverage.py: This is a private format, don't read it directly!{"lines":{"/home/rohit/projects/IRCLogParser/test/tests.py":[],"/home/rohit/projects/IRCLogParser/test/unit_tests.py":[1,2,3,4,5,6,7,10,11,12,15,16,17,19,20,22,23,25,26,27,28,29,31,32,34,35,36,38,40,41,42,43,44,46,47,49,50,51,52,53,54,55,57,58,59,60,61,62,64,65,66,67,69,70,72,73,75,76,77,78,79,80,82,83,84,85,87,88,89,90,91,94,95,96,97 ...
```
please put .coverage file in .gitignore list.

prasadtalasila reviewed (changes requested):
please make the suggested changes.

prasadtalasila commented:
@kaushik-rohit The tests/slack/test_units.py file should be renamed to tests/slack/test_util.py.
